### PR TITLE
[Keycloak] Identity Provider module mappers fix

### LINKED ISSUE #6002: [Keycloak] Identity Provider unchanged mappers are deleted
### Summary

When updating an IdP with a deterministic list of mappers, unchanged mappers are deleted because no changes are detected.

### Issue Type

Bug Report

### Component Name

keycloak_identity_provider

### Ansible Version

```console (paste below)
$ ansible --version
ansible [core 2.14.2]
  config file = [...]/ansible.cfg
  configured module search path = ['[...]/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = [...]/venv/lib/python3.10/site-packages/ansible
  ansible collection location =[...]/{{ ANSIBLE_HOME ~ "/collections" }}
  executable location = [...]/venv/bin/ansible
  python version = 3.10.9 (main, Dec 19 2022, 17:35:49) [GCC 12.2.0] ([...]/venv/bin/python)
  jinja version = 3.1.2
  libyaml = True
```

### Community.general Version

```console (paste below)
$ ansible-galaxy collection list community.general
Collection        Version
----------------- -------
community.general 6.3.0
```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed

```

### OS / Environment

_No response_

### Steps to Reproduce

Given email is already created and given_name is to be created, here is an example mappers attribute for keycloak_identity_provider module:
```yaml (paste below)
    mappers:
    - name: email
      identityProviderMapper: oidc-user-attribute-idp-mapper
      config:
        claim: email
        user.attribute: email
        syncMode: FORCE
    - name: first_name
      identityProviderMapper: oidc-user-attribute-idp-mapper
      config:
        claim: given_name
        user.attribute: firstName
        syncMode: FORCE
```

### Expected Results

All mappers are present in Keycloak IdP if only one exists

### Actual Results

Existing mapper is deleted 

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by ymazzer**
##### SUMMARY
Existing mappers unchanged in desired state were being deleted because absent of changed map. Fixed the behavior by considering desired_idp instead of changed mappers list for deletion.

Fixes #6002

##### ISSUE TYPE
- Bugfix Pull Request
- 
##### COMPONENT NAME
keycloak_identity_provider

### CODE REVIEW

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution. Could you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

**Comment by ymazzer** [comment]
> Thanks for your contribution. Could you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

This is done :)

File: changelogs/fragments/6002-keycloak-identity-provider-mappers-deterministic-changes [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
```
Comment (felixfontein) [inline changelogs/fragments/6002-keycloak-identity-provider-mappers-deterministic-changes:1]: Please add a `.yml` extension to this file :)

Comment (ymazzer) [reply changelogs/fragments/6002-keycloak-identity-provider-mappers-deterministic-changes]: Oups sorry my bad.

This is fixed.

**Comment by felixfontein** [comment]
Please note that the unit tests are also failing.

**Comment by felixfontein** [comment]
ping @ymazzer 

needs_info

**Pull request closed without merging**